pkg/controller/directvolumemigrationprogress/directvolumemigrationprogress_controller.go

```diff
@@ -347,7 +347,8 @@ func (r *RsyncPodProgressTask) getRsyncClientContainerStatus(podRef *kapi.Pod, p
 		LogMessage:        podRef.Status.Message,
 	}
 	var containerStatus *kapi.ContainerStatus
-	for _, c := range podRef.Status.ContainerStatuses {
+	for i := range podRef.Status.ContainerStatuses {
+		c := podRef.Status.ContainerStatuses[i]
 		if c.Name == RsyncContainerName {
```

Comment:
Why is this change needed?

Reply:
I mean two lines above

Reply:
So this piece of code was taking the reference of the index variable and not the actual status reference (was not observed earlier because there was only on container in the Pod). This was identified by @pranavgaikwad while he was testing out the PR.